base/redactor.go

```diff
@@ -15,8 +15,10 @@ type Redactor interface {
 	// changed, hashed, or removed completely depending on desired behaviour.
 	Redact() string
 }
+type RedactorFunc func() Redactor
 
 type RedactorSlice []Redactor
+type RedactorFuncSlice []RedactorFunc
```

Comment:
Like discussed below in the other comment - Since RedactorFunc implements Redactor, we don't need a new type.